File: WeatherWidget/src/main/java/org/voegtle/weatherwidget/DiagramFragmentPagerAdapter.java
package org.voegtle.weatherwidget;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;

public class DiagramFragmentPagerAdapter extends FragmentPagerAdapter {

  private ArrayList<DiagramFragment> fragments = new ArrayList<DiagramFragment>();

  public DiagramFragmentPagerAdapter(FragmentManager fm) {
    super(fm);
  }

  public void add(DiagramFragment fragment) {
    fragments.add(fragment);
  }

  @Override
  public DiagramFragment getItem(int i) {
    return fragments.get(i);
  }

  @Override
  public int getCount() {
    return fragments.size();
  }
}

File: src/main/java/org/voegtle/weatherwidget/diagram/DiagramFragmentPagerAdapter.kt
package org.voegtle.weatherwidget.diagram

import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import java.util.ArrayList

class DiagramFragmentPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

  private val fragments = ArrayList<DiagramFragment>()

  fun add(fragment: DiagramFragment) {
    fragments.add(fragment)
  }

  override fun getItem(i: Int): DiagramFragment {
    return fragments[i]
  }

  override fun getCount(): Int {
    return fragments.size
  }
}
